### PR TITLE
Add ErrorData and HeaderRenderer classes

### PR DESCRIPTION
**Pull request by devanych**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | #12, #13

### CODE REVIEW

File: src/ErrorData.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+ */
+final class ErrorData
+{
+    /**
+     * @var string The content of error data.
+     */
+    private string $content;
+
+    /**
+     * @var array<string, string|string[]> The headers of error data to add to the response.
```
Comment (samdark) [inline src/ErrorData.php:20]:
```suggestion
     * @var array<string, string|string[]> The headers to add to the response.
```

File: src/ErrorData.php [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+/**
+ * ErrorData stores content and headers that are suitable for displaying.
+ */
+final class ErrorData
+{
+    /**
+     * @var string The content of error data.
+     */
+    private string $content;
```
Comment (samdark) [inline src/ErrorData.php:17]:
```suggestion
    /**
     * @var string The content to use as response body.
     */
    private string $content;
```

File: src/ErrorData.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * @var array<string, string|string[]> The headers of error data to add to the response.
+     */
+    private array $headers;
+
+    /**
+     * @param string $content The content of error data.
+     * @param array<string, string|string[]> $headers The headers of error data to add to the response.
```
Comment (samdark) [inline src/ErrorData.php:26]:
```suggestion
     * @param string $content The content to use as response body.
     * @param array<string, string|string[]> $headers The headers to add to the response.
```

File: src/ErrorData.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        $this->content = $content;
+        $this->headers = $headers;
+    }
+
+    /**
+     * Returns a content of error data.
+     *
+     * @return string The content of error data.
+     */
+    public function __toString(): string
```
Comment (samdark) [inline src/ErrorData.php:39]: Do we need this method?

File: src/Middleware/ErrorCatcher.php [hunk 5]
```diff
@@ -59,43 +62,45 @@ public function __construct(
...
 
         $new = clone $this;
-        $new->renderers[$this->normalizeMimeType($mimeType)] = $rendererClass;
+        $new->renderers[$this->normalizeContentType($contentType)] = $rendererClass;
         return $new;
     }
 
     /**
-     * @param string[] $mimeTypes MIME types or, if not specified, all will be removed.
+     * @param string[] $contentTypes MIME types or, if not specified, all will be removed.
```
Comment (samdark) [inline src/Middleware/ErrorCatcher.php:75]:
```suggestion
     * @param string[] $contentTypes MIME types to remove associated renderers for. If not specified, all renderers will be removed.
```

File: src/ThrowableRendererInterface.php [hunk 6]
```diff
@@ -8,27 +8,29 @@
 use Throwable;
 
 /**
- * ThrowableRendererInterface converts throwable into its string representation
+ * ThrowableRendererInterface converts throwable into error data suitable for displaying.
```
Comment (samdark) [inline src/ThrowableRendererInterface.php:11]:
```suggestion
 * ThrowableRendererInterface converts throwable into error data suitable for adding it to response.
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/ErrorData.php (hunk 4 above)
Comment (devanych) [reply src/ErrorData.php:39]: Of course you do. It is also used in `ErrorHandler`, which returns a string, not a response instance.

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
👍

**Pull request merged**